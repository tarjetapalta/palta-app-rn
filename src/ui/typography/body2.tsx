import React, { FunctionComponent } from 'react';

import { Text, TextProps } from './base';

export const Body2: FunctionComponent<
  Pick<TextProps, 'testID' | 'color' | 'emphasis' | 'textAlign'>
> = ({
  /** Test id for test */
  testID,
  /** Text */
  children,
  /** Text color */
  color,
  /** Text font opacity */
  emphasis,
  /** Text align */
  textAlign,
}) => (
  <Text
    testID={testID}
    color={color}
    size="xs"
    weight="regular"
    emphasis={emphasis}
    textAlign={textAlign}
  >
    {children}
  </Text>
);
