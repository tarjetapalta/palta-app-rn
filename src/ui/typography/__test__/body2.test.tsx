import React from 'react';
import renderer from 'react-test-renderer';

import WithTheme from '../../../theme/context';
import { Body2 } from '../body2';

describe('body2 component', () => {
  it('basic Body2 with color "text", emphasis "high" and testID "Body2"', () => {
    const result = renderer
      .create(
        <WithTheme>
          <Body2>Some text</Body2>
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
  it('basic Body2 with color "primary", emphasis "medium"', () => {
    const result = renderer
      .create(
        <WithTheme>
          <Body2 color="primary" emphasis="medium">
            Some text
          </Body2>
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
  it('basic with custom testId', () => {
    const result = renderer
      .create(
        <WithTheme>
          <Body2 testID="custom testID">Some text</Body2>
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
});
