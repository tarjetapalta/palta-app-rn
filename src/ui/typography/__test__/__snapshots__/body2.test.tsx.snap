// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`body2 component basic Body2 with color "primary", emphasis "medium" 1`] = `
<Text
  color="primary"
  emphasis="medium"
  numberOfLines={0}
  size="xs"
  style={
    Array [
      Object {
        "color": "#91b086",
        "fontFamily": "BwModelica-Regular",
        "fontSize": 12,
        "opacity": 0.6,
        "textAlign": "center",
      },
    ]
  }
  weight="regular"
>
  Some text
</Text>
`;

exports[`body2 component basic Body2 with color "text", emphasis "high" and testID "Body2" 1`] = `
<Text
  numberOfLines={0}
  size="xs"
  style={
    Array [
      Object {
        "color": "#283238",
        "fontFamily": "BwModelica-Regular",
        "fontSize": 12,
        "opacity": 0.87,
        "textAlign": "center",
      },
    ]
  }
  weight="regular"
>
  Some text
</Text>
`;

exports[`body2 component basic with custom testId 1`] = `
<Text
  numberOfLines={0}
  size="xs"
  style={
    Array [
      Object {
        "color": "#283238",
        "fontFamily": "BwModelica-Regular",
        "fontSize": 12,
        "opacity": 0.87,
        "textAlign": "center",
      },
    ]
  }
  testID="custom testID"
  weight="regular"
>
  Some text
</Text>
`;
